extensions/vscode/src/utils/vscode.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2024 by Posit Software, PBC.
```

Comment:
Related to pushing on having more tests on the extension code, this file contains a few methods for which it is worth having tests for.

Reply:
Will be adding tests.

Reply:
I investigated implementing `vitests` for this utility, using a few different techniques, but all were unsuccessful. Attempts to run the tests failed as they attempted to resolve the `vscode` imports, in which they could not resolve the type definition into an import. 

We also discussed the two techniques I investigated (mocking the extension runtime environment and injection pattern for removing the extension runtime dependencies). We concluded that going this route would be very costly to mock the entire runtime environment of VSCode extension and do it correctly where it would accurately represent the actual runtime environment.

Instead, we quickly looked back at the guidelines provided by Microsoft on how to best test the extensions. (https://code.visualstudio.com/api/working-with-extensions/testing-extension). 

We already have the [Extension Test Runner](https://marketplace.visualstudio.com/items?itemName=ms-vscode.extension-test-runner) setup in our project, with one simple test. It is executed with `npm run test`. While there are some changes that we'll probably want to make (like switching from `mocha` to `jest`), this appears to follow the best practices for testing functionality within the extension's runtime environment.

I propose that we go in this direction and that it is outside of the scope of this PR. I'd recommend we move forward with this PR, merge it, and then implement some simple tests for the detection of the preferred R and Python versions using the extension test runner.

@dotNomad I think this summarizes our discussion?

@marcosnav Let's discuss at our next opportunity, if you have any questions or concerns.

Reply:
Great write up @sagerb thanks for putting that down in a comment.